### ISSUE TITLE
"Latest Releases" links are broken

### ISSUE THREAD

**Issue by brianhigh** [WGSExtract/WGSExtract.github.io#27]
"Latest Releases" links on https://wgsextract.github.io/ are broken. I am getting a JSON error in my web browser when I click on any of them.

Parsed:

```
error	
code	"invalidRequest"
message	"The provided share id was malformed."
```

Raw:

```
{"error":{"code":"invalidRequest","message":"The provided share id was malformed."}}
```

**Comment by RandyHarr** [comment]
Odd. Appears MS Onedrive / Azure may have changed their link format. Will investigate and update. Traveling this week but will try and share working links if I get a chance to generate.

**Comment by brianhigh** [comment]
Thanks. :) Relatedly, I presume, there are issues when running the installer (which I downloaded a couple of days ago). Before the MS cloud problem started, this installer (WGSExtract-Dev_latest_installer.zip) worked fine for me (Ubuntu 24.04 LTS) but now it is broken when testing on other machines. Specifically, after the dependencies (e.g., system libraries and utilities) get installed, there are problems downloading the "WGS Extract Installer" and later downloads. Tested on Ubuntu 24.04 LTS and macOS Sequoia 15.1.1. Here is some output, abridged.

```
..  Started upgrading WGS Extract Installer from v50 ...
curl: (6) Could not resolve host: null
Warning: Problem : timeout. Will retry in 1 seconds. 5 retries left.
```

And then...

```
*** Retrieving the WGS Extract latest release json file (track Dev).
############################################################################################ 100.0%

*** WGS Extract Installer v50 is installed but outdated.

...  Started installing WGS Extract Reference Library vnull ...
curl: (6) Could not resolve host: null
Warning: Problem : timeout. Will retry in 1 seconds. 5 retries left.
```

And same for WGS Extract Local Tools, WGS Extract Program, etc.

Again, this is likely related to the MS cloud issue -- if that's where these are stored.

**Comment by RandyHarr** [comment]
You are correct. The links we use bypass the web interface dialogs and allow direct retrieval via code and scripts.  Seems MS has disabled the onedrive links without any notice or warning to others.  Am trying to discover the new workaround as they are needed to download the installers and for the installer to download the rest of the packages.  Give me a day or so to return from traveling and dig into it.

Historically, you could change a provided share link by taking out the file-unique code and pasting into the template:
https://api.onedrive.com/v1.0/shares/UNIQUE_FILE_CODE/root/content

**Comment by RandyHarr** [comment]
It appears MS has turned off direct download URLs for Onedrive in the API.  You can still get it on much more expensive Business Sharepoint services with egress fees. I am looking for an alternative cloud provider now that Google and MS have turned this vital feature off for lower cost "personal" accounts. Checking out Wasabi and Mega now. Once fixed, just requires editing a single file to get everything switched over to the new service. My hosting provider does not do well with serving the large files needed for installs. And I have not switched back to my own servers for 20+ years for reliability.

**Comment by RandyHarr** [comment]
Should be back up and operational now.  

In the end, only had to edit a single file with a hundred or so file redirects. A setup designed when Google Drive stopped allowing direct downloads of files >25 MB.  And we had to switch from Google Drive to MS Onedrive. But it was troublesome finding a cloud service to host the files with our current code.

Onedrive turned off direct file download in their API interface as is needed by programs. Sharepoint looked like a good alternative BUT they do not support / allow the use of curl. We use curl as it is the only common program available on all platforms without install.  We need it to bootstrap and get the other modules. Wasabi and others, while good cloud services, were simply going to require too much work to integrate in.  So we decided to move the files to our web server.  Not as fast and reliable as cloud file servers; but will have to do for now.
